src/runtime/virtcontainers/clh.go

```diff
@@ -1304,6 +1476,11 @@ func (clh *cloudHypervisor) addVSock(cid int64, path string) {
 	}).Info("Adding HybridVSock")
 
 	clh.vmconfig.Vsock = chclient.NewVsockConfig(cid, path)
+
+	if clh.config.BootToBeTemplate || clh.config.BootFromTemplate {
+		id := "_vsock0"
+		clh.vmconfig.Vsock.Id = &id
+	}
```

Comment:
Can you please explain why explicit name is needed here?

Reply:
Thanks for your review. If not explicit name, BootToBeTemplate  vm's Vsock.Id is _vsock0, but BootFromTemplate vm's Vsock.Id is _vsock1, this will cause restore to fail。

Reply:
Thank you for the additional information. That's quite strange to me. The snapshot/restore functionality of Cloud Hypervisor does not require users to set the device `Id` explicitly.  

I believe the issue is related to again my other questions about why we are relying on internal information from Cloud Hypervisor to support VM templating in Kata (e.g. config.json/stage.json files of CLH snapshot folder).

Reply:
Each vm has a different vsock path,  so each BootFromTemplate vm's config.json is different.